### ISSUE TITLE
What should I do?

### ISSUE THREAD

**Issue by iRYO400** [diareuse/mCache#2]
Cache.give(myObj)
          .build()
          .getNow();

give method is red and:
```
Error: non-static method <T>give(T) cannot be referenced from a static context
where T is a type-variable:
T extends Object declared in method <T>give(T)
```

**Comment by diareuse** [comment]
You are using Java, correct?

**Comment by iRYO400** [comment]
Yep!

**Comment by diareuse** [comment]
ok, let me investigate this further for you.

in the meantime, you can create kotlin wrapper for Cache class so you can keep using mCache

**Comment by iRYO400** [comment]
specifically for only Cache class? so how to do it?

**Comment by diareuse** [comment]
You can make kotlin wrapper like so:
```
object CacheWrapper {

  @JvmStatic
  fun <T : Any> obtain(cls: Class<T>): FilePresenterBuilder<T> {
    return Cache.obtain(cls)
  }

  @JvmStatic
  fun <T : Any> give(file: T): FilePresenterBuilder<T> {
    return Cache.give(file)
  }

}
```

Compiled kotlin will respect functions **give(t: T)** and **obtain(cls: Class<T>)** as static functions.

**Comment by diareuse** [comment]
\* make sure you are including this in you app-level **build.gradle**:
```
dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
}
```

and

```
buildscript {
    ext.kotlin_version = '1.1.51'
    ...
    dependencies {
        ...
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    }
}
```

in you project-level **build.gradle**.


FYI Android Studio 3 does this for you. 😃

**Comment by diareuse** [comment]
I noticed you are using **.getNow()** method; make sure you are using this off main thread & read docs 😃, you can ommit using **.getLater()** for small files, however it's great practice using async calls all the time though 👍 

I'll push fix for this issue as soon as I can!

**Comment by diareuse** [comment]
Fixed in 0874d68
